pysat/tests/test_instrument.py

```diff
@@ -382,42 +394,58 @@ def test_repr_w_orbit(self):
                                     update_files=True,
                                     orbit_info=orbit_info)
 
-        output = testInst.__str__()
+        self.out = testInst.__str__()
         # Check that orbit info is passed through
-        assert output.find('Orbit properties not set') < 0
-        assert output.find('Orbit Kind:') > 0
-        assert output.find('Loaded Orbit Number: None') > 0
+        assert self.out.find('Orbit properties not set') < 0
+        assert self.out.find('Orbit Kind:') > 0
+        assert self.out.find('Loaded Orbit Number: None') > 0
         # Activate orbits, check that message has changed
-        testInst.load(2009, 1)
+        testInst.load(self.ref_time.year, self.ref_doy)
         testInst.orbits.next()
-        output = testInst.__str__()
-        assert output.find('Loaded Orbit Number: None') < 0
-        assert output.find('Loaded Orbit Number: ') > 0
+        self.out = testInst.__str__()
+        assert self.out.find('Loaded Orbit Number: None') < 0
+        assert self.out.find('Loaded Orbit Number: ') > 0
 
-    def test_repr_w_padding(self):
+    def test_str_w_padding(self):
+        """Test string output with data padding """
         self.testInst.pad = pds.DateOffset(minutes=5)
-        output = self.testInst.__str__()
-        assert output.find('DateOffset: minutes=5') > 0
+        self.out = self.testInst.__str__()
+        assert self.out.find('DateOffset: minutes=5') > 0
 
-    def test_repr_w_custom_func(self):
+    def test_str_w_custom_func(self):
+        """Test string output with custom function """
         def testfunc(self):
             pass
         self.testInst.custom.attach(testfunc, 'modify')
-        output = self.testInst.__str__()
-        assert output.find('testfunc') > 0
-
-    def test_repr_w_load_data(self):
-        self.testInst.load(2009, 1)
-        output = self.testInst.__str__()
-        assert output.find('No loaded data') < 0
-        assert output.find('Number of variables:') > 0
-        assert output.find('uts') > 0
+        self.out = self.testInst.__str__()
+        assert self.out.find('testfunc') > 0
+
+    def test_str_w_load_lots_data(self):
+        """Test string output with loaded data """
+        self.testInst.load(self.ref_time.year, self.ref_doy)
+        self.out = self.testInst.__str__()
+        assert self.out.find('Number of variables:') > 0
+        assert self.out.find('uts') < 0
+
+    def test_str_w_load_less_data(self):
+        """Test string output with loaded data """
+        # Load the test data
+        self.testInst.load(self.ref_time.year, self.ref_doy)
+
+        # Ensure the desired data variable is present and delete all others
+        self.testInst.data = self.testInst.data[['mlt']]
```

Comment:
```suggestion
        # minimum of 4 variables needed to test all lines
        self.testInst.data = self.testInst.data[self.testInst.variables[0:4]]
```

Reply:
Thanks for figuring out the reason why this was going wrong!  The suggested change isn't sufficient to fix the test, so I committed something more complete below.